### PR TITLE
Feat/POST github, GET meme routes

### PR DESCRIPTION
**Pull request by rohanjoby-ws**
### Ticket Link

---

### Related Links

---

### Description
- add POST route for GitHub to get user details
- add tests for GitHub route
- add get memes route
- add tests for get memes route
---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: functions/github-post-function/tests/index.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should call the github-post-function', async () => {
+		mockGetUserData = jest
+			.spyOn(github, 'getUser')
+			.mockImplementation(Promise.resolve(MOCK_RESPONSE));
+		// mock return value
+		mockGetUserData.mockReturnValue(Promise.resolve(MOCK_RESPONSE));
```
Comment (vishnu-wednesday) [inline functions/github-post-function/tests/index.test.js:30]: I think this is redundant. mockImplementation creates a implementation that will be called. You can just use that right?  Do we also need a return value too?

File: resources/lambda/functions.yml [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+githubPostFunction:
+  handler: functions/github-post-function/index.handler
   role: ServerlessTemplateRole
   events:
     - http:
-        method: GET
-        path: /get-repos
+        method: POST
+        path: /get-user
         cors: true
```
Comment (vishnu-wednesday) [inline resources/lambda/functions.yml]: Do we need cors?

**Review by vishnu-wednesday — COMMENTED** [review]
Looks great to me! Good Job!

File: resources/lambda/functions.yml (hunk 2 above)
Comment (rohanjoby-ws) [reply resources/lambda/functions.yml]: Ohk yes, cors is not needed here. Will make the change

File: functions/github-post-function/tests/index.test.js (hunk 1 above)
Comment (rohanjoby-ws) [reply functions/github-post-function/tests/index.test.js:30]: In the current implementation, it is needed,
but I can just mockReturnValue without a mockImplementation

Comment (rohanjoby-ws) [reply functions/github-post-function/tests/index.test.js:30]: In the current implementation, it is needed, without it `specificMockImpl.apply` is not a function error is thrown for `getUser`

**Pull request merged**